templates/base/client/.rtag/App.tsx.hbs

```diff
@@ -0,0 +1,103 @@
+import React, { useEffect, useState } from "react";
+import ReactDOM from "react-dom";
+import { ToastContainer, Zoom, toast } from "react-toastify";
+{{#if auth.google}}
+import { GoogleLogin } from "react-google-login";
+{{/if}}
+import { RtagClient, RtagConnection } from "./client";
+import { {{userState}} as UserState } from "./types";
+import { RtagContext } from "./context";
+import { Forms, InitializeForm } from "./Forms";
+import { State } from "./State";
+
+const client = new RtagClient(import.meta.env.VITE_APP_ID);
+
+function App() {
+  const [connection, setConnection] = useState<RtagConnection>();
+  const [state, setState] = useState<UserState>();
+  const [token, setToken] = useState<string | null>(sessionStorage.getItem("token"));
+
+  useEffect(() => {
+    const path = location.pathname;
+    if (path.startsWith("/state/") && token !== null) {
+      const stateId = path.split("/state/").pop()!;
```

Comment:
nit: i think you should handle this pop() => null case instead of using ! because it could technically be possible if they delete the state ID from the path